### ISSUE TITLE
Implement partial readout

### ISSUE THREAD

**Issue by SaraRegibo** [IvS-KULeuven/PlatoSim3#285]
Basic concept:
* block of rows: parallel + serial readout
* rows below (i.e. closer to the readout register): parallel readout only
* rows above (i.e. further away from the readout register): parallel readout only

**Comment by SaraRegibo** [comment]
Proposed structure in the configuration file:

```
CCD:
        ReadoutMode:
                ReadoutMode:       Nominal/Partial
                Partial:
        	     FirstRowReadout:    0
        	     NumRowsReadout:     100

	SerialTransferTime:           340
	ParallelTransferTime:         110
        ParallelTransferTimeFast:     90
```

This way, we can configure all readout modes (normal/fast nominal/partial) in the same way:

* normal camera in nominal mode: 4510 rows starting at row 0
* normal camera in partial readout model: arbitrary block of complete rows somewhere on the detector
* fast camera in nominal mode: 2255 rows starting at row 2255 (lower half of the detector shielded off against incoming flux)
* fast camera in partial readout mode: arbitrary block of complete rows somewhere on the upper half of the detector

**Comment by SaraRegibo** [comment]
Crucial things about the fast cameras to take into account:
* lower half of the detector is shielded off against incoming flux
* when the lower half is being read out, the rows of the upper half stay put

**Comment by SaraRegibo** [comment]
# Effects that may have to be adapted

* `takeExposure`: advance internal time by the duration of the readout that happens before the next exposure starts:
     - normal cam: readout of the complete CCD
     - fast cam: frame transfer (readout of the lower half of the CCD after frame transfer happens while the next exposure is already taking place)
* dark signal:
     - dark signal accumulated during exposure + whole readout
     - partial readout: no smearing map
     - smearing map for fast cam in nominal mode: dark signal accumulated during the readout of the lower half of the CCD (after charge transfer)
* photon noise:
     - partial readout: no smearing map
* cosmics:
     - partial readout: no smearing map
     - image area: cosmic hits during exposure + whole readout process
     - smearing map for fast cam in nominal mode: cosmic hits during the readout of the lower half of the CCD (after charge transfer)
     - bias map: cosmic hits during readout of entire readout register = (2255 + numColumnsBiasMap) * 110µs
* CTI:
     - difference in parallel transfer time only due to delays (for clock settling) => no change in configuration parameters
     - Short -> adapt readout time
* open-shutter smearing
     - partial readout: no smearing map
     - partial readout => faster => less open-shutter smearing
* aberration correction (camera): `timeMiddle`
* advance internal time for jitter
* advance internal time for drift
* advance internal time in `Simulation`

# Unchanged

* BFE
* full-well saturation (blooming)
* readout noise
* gain
* electronic offset
* digital saturation
* flatfield
* convolution
* re-binning
* throughput efficiency
* charge diffusion / jitter smoothing

**Comment by SaraRegibo** [comment]
Closed by pull request #296
